Apps/Medication/src/Parsers/TRPParser.cs

```diff
@@ -0,0 +1,130 @@
+﻿//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.MedicationService.Parsers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using HealthGateway.MedicationService.Models;
+    using HL7.Dotnetcore;
+    using Microsoft.Extensions.Configuration;
+
+    /// <summary>
+    /// Parser of TRP (Patient Profile) messages.
+    /// </summary>
+    public static class TrpParser
```

Comment:
I dont believe this should be static. Instead the parser could be created on the fly as a worker and get dispose or we could set it as a dependency injection singleton.